plugins/modules/net_tools/nmcli.py

```diff
@@ -1288,6 +1341,26 @@
     autoconnect: true
     state: present
 
+- name: Create a VPN L2TP connection for ansible_user to connect on vpn.example.com authenticating with user 'brittany' and pre-shared key as 'Brittany123'
+  block:
+    - ansible.builtin.set_fact:
+        psk: "0s{{ 'Brittany123' | ansible.builtin.b64encode }}"
+
+    - name: Create the connection
+      community.general.nmcli:
+        type: vpn
+        conn_name: my-vpn-connection
+        vpn:
+            permissions: "{{ ansible_user }}"
+            service-type: l2tp
+            gateway: vpn.example.com
+            password-flags: 2
+            user: brittany
+            ipsec-enabled: true
+            ipsec-psk: "0s{{ 'Brittany123' | ansible.builtin.b64encode }}"
+        autoconnect: false
+        state: present
```

Comment:
The block and set_facts are no longer needed:
```suggestion
  community.general.nmcli:
    type: vpn
    conn_name: my-vpn-connection
    vpn:
        permissions: "{{ ansible_user }}"
        service-type: l2tp
        gateway: vpn.example.com
        password-flags: 2
        user: brittany
        ipsec-enabled: true
        ipsec-psk: "0s{{ 'Brittany123' | ansible.builtin.b64encode }}"
    autoconnect: false
    state: present
```